### DISCUSSION TITLE
give players a choice between opengl and Vulcan

### DISCUSSION THREAD

**Discussion by Nepil38** [skyline-emu/skyline#1815] [Ideas]
give players a choice between opengl and Vulcan

**Comment by dima-xd** [comment]
How you can have a choice between opengl and vulkan if skyline will use only vulkan?

**Reply by Nepil38** [reply]
I said to give players a choice between opengl or vulkan, as it is done in yuzu, that is, add the ability

**Comment by Swizzlefowl** [comment]
Skyline will never implement opengl as it does not have the necessary features that Skyline requires.

**Reply by Nepil38** [reply]
then tell me what open gl functions are not available to prevent players from choosing the type of render that suits them

**Reply by iniMerchant64** [reply]
OpenGL ES is pretty much too barebones. Its not good enough to handle something as complex as skyline. I remind you that arm versions of APIs are inferior to their x86 counterparts

**Reply by dima-xd** [reply]
> then tell me what open gl functions are not available to prevent players from choosing the type of render that suits them

Opengl is garbage. That's it

**Reply by Nepil38** [reply]
Well, listen, I can say the same about vulkan. i.e. you mean that skyline will not be able to work with opengl while the yuzu emulator works fine with opengl on pc

**Reply by dima-xd** [reply]
> Well, listen, I can say the same about vulkan. i.e. you mean that skyline will not be able to work with opengl while the yuzu emulator works fine with opengl on pc

Skyline != Yuzu. And to implement opengl api usage in skyline will take a lot of time

**Reply by Nepil38** [reply]
Well, it would be cool to have a similar function in the future

**Reply by iniMerchant64** [reply]
> Well, listen, I can say the same about vulkan. i.e. you mean that skyline will not be able to work with opengl while the yuzu emulator works fine with opengl on pc

Yet again, opengl es is opnegls inferior and cut down version. It is not usable in an emulator this complex

**Reply by iniMerchant64** [reply]
> Well, it would be cool to have a similar function in the future

Wont happen. We know as much

**Comment by Swizzlefowl** [comment]
That's opengl yes what mobile devices have is called opengles and it's a very barebones version of opengl and just not suited for skyline.

**Comment by PixelyIon** [comment]
The version of OpenGL on Android is OpenGL ES (OpenGL for Embedded Systems), this is significantly cut-down from the desktop version and doesn't have an adequate amount of features for Switch emulation. Even if this wasn't the case, OGLES drivers from mobile vendors are fairly buggy due to them inherently needing to do far more in the driver compared to Vulkan which moves a lot of the burden onto the API user (Skyline) rather than implementer (GPU driver developer).
